File: sort/selectionsort.c
#include <stdio.h>

void selectionsort(int *p, size_t size);
void print_array(int *p, size_t size);

int main() {
  int a[13] = { 3, 2, 12, 4, 22, 32, 44, 444, 12, 65, 9, -2, -13 };

  print_array(a, 13);
  selectionsort(a, 13);
  print_array(a, 13);

  return 0;
}

void print_array(int *p, size_t size) {
  int i;
  printf("\nArray is: ");
  for (i = 0; i < size; ++i) {
    printf("%d ", p[i]);
  }
}

void selectionsort(int *p, size_t size) {
  int i, j, temp, min_index;

  for (i = 0; i < size; i++) {
    min_index = i;
    for (j = i + 1; j < size; ++j) {
      if (p[j] < p[i]) {
        min_index = j;   
      }
    }

    temp = p[i];
    p[i] = p[min_index];
    p[min_index] = temp;
  }
}

File: sort/selectionsort.py
def selectionsort(A):
  for i in range(len(A)):
    for j in range(i + 1, len(A)):
      if A[i] > A[j]:
        temp = A[i]
        A[i] = A[j]
        A[j] = temp
  
  return A


print selectionsort([3, 2, 5, 13, 0])
print selectionsort([-3, -2, 5, 163, 0])
print selectionsort([3, 2, 15, 413, 500])
print selectionsort([390, -52, 5, -13, 0])
